io/net/virt-net/network_virtualization.py

```diff
@@ -380,6 +424,61 @@ def ping_check(self):
             return False
         return True
 
+    def trigger_failover(self, device):
+        '''
+        Triggers failover from HMC
+        '''
+        backing_dev_logport_id = self.get_backing_device_logport()
+        network_dev_logport_id = self.get_device_logport()
+        if device == 'backing_device':
+            cmd = 'chhwres -r virtualio --rsubtype vnicbkdev -o act -m %s \
+                   -s %s --id %s --logport %s' % (self.server, self.slot_num,
+                                                  self.lpar_id,
+                                                  backing_dev_logport_id)
+        else:
+            cmd = 'chhwres -r virtualio --rsubtype vnicbkdev -o act -m %s \
+                   -s %s --id %s --logport %s' % (self.server, self.slot_num,
+                                                  self.lpar_id,
+                                                  network_dev_logport_id)
+        try:
+            cmd = self.run_command(cmd)
+        except CommandFailed as cmdfail:
+            self.log.debug(str(cmdfail))
+            self.fail("Command to set %s as Active has failed" % device)
+
+    def get_backing_device_logport(self):
+        '''
+        Get the logical port id of the
+        backing device
+        '''
+        output = self.backing_dev_list()
+        for i in output:
+            if i.startswith('%s,' % self.slot_num):
+                logport = i.split(',')[1:][1].split('/')[1]
+        return logport
+
+    def get_device_logport(self):
+        '''
+        Get the logical port id of the Network
+        virtualized device
+        '''
+        output = self.backing_dev_list()
+        for i in output:
+            if i.startswith('%s,' % self.slot_num):
+                device_logport = i.split(',')[0:][1].split('/')[1]
+        return device_logport
```

Comment:
Will it return only one device logport or an array (if there are more devices) ?